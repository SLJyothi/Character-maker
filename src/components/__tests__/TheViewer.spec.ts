/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallowMount, VueWrapper } from '@vue/test-utils';
import TheViewer from '../TheViewer.vue';

describe('The Viewer', () => {
  let wrapper: VueWrapper<any>;
  window.scrollTo = function(): void {};
  beforeEach(() => {
    wrapper = shallowMount(TheViewer, {
      global: {
        mocks: {
          window
        },
        stubs: ['the-character']
      }
    });
  });
  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should change download text when downloadSVGasPNG is called', () => {
    wrapper.vm.downloadSVGAsPNG();
    expect(wrapper.vm.downloadText).toBe('Loading...');
  });

  it('should create an anchor tag with src value equals to canvas dataurl', () => {
    const canvas = document.createElement('canvas');
    wrapper.vm.downloadCanvas(canvas);
    expect(wrapper.find('canvas')).toBeTruthy();
  });
});
