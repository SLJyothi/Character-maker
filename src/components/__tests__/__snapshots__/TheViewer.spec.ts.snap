// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The Viewer > should match the snapshot 1`] = `
<section
  class="viewer"
  data-v-523bb9d8=""
>
  <section
    class="viewer__credit"
    data-v-523bb9d8=""
  >
    <p
      data-v-523bb9d8=""
    >
       Developed with ♥️ by 
      <a
        class="viewer__credit-link"
        data-v-523bb9d8=""
        href="https://github.com/SLJyothi"
      >
         Jyothi SL 
      </a>
    </p>
  </section>
  <div
    class="viewer__avatar"
    data-v-523bb9d8=""
  >
    <the-character-stub
      data-v-523bb9d8=""
    />
  </div>
  <section
    class="viewer__credit"
    data-v-523bb9d8=""
  >
    <p
      data-v-523bb9d8=""
    >
       Create for Fun 
    </p>
  </section>
  <button
    class="viewer__btn"
    data-v-523bb9d8=""
    type="button"
  >
    Download Image
  </button>
</section>
`;
